runtime/dynlink_nat.c

```diff
@@ -82,12 +82,14 @@ CAMLprim value caml_natdynlink_open(value filename, value global)
   void *sym;
   void *dlhandle;
   char_os *p;
+  int global_dup;
 
   /* TODO: dlclose in case of error... */
 
   p = caml_stat_strdup_to_os(String_val(filename));
+  global_dup = Int_val(global);
```

Comment:
How about [this](https://github.com/OlivierNicole/ocaml/compare/fix_data_races...smuenzel:ocaml:fix_data_races) instead? It has the advantage of not introducing a `dup` variable, which we aren't doing elsewhere.

Reply:
It goes against [Rule 1](https://v2.ocaml.org/manual/intfc.html#ss:c-simple-gc-harmony), but if it’s done elsewhere, or if people are fine with not strictly applying the rules in compiler-internal libraries, I won’t fight it.

Reply:
This sounds fine indeed as the runtime follows relaxed rules. This could be strengthened with a `CAMLassert(Is_long(global))` for readability in place of the comment.